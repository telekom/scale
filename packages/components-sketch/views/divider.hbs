{{!--
Scale https://github.com/telekom/scale

Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG

This Source Code Form is subject to the terms of the Mozilla Public
License, v. 2.0. If a copy of the MPL was not distributed with this
file, You can obtain one at https://mozilla.org/MPL/2.0/.
--}}

<div data-sketch-artboard="Divider" class="artboard">
  <h1 class="title--artboard">Divider</h1>

  <h2 class="title--artboard-section">Standard</h2>
  <div class="agent-states--fullwidth">
    <scale-divider data-sketch-key="divider-horizontal"></scale-divider>
  </div>

  <h2 class="title--artboard-section">Vertical</h2>
  <div class="agent-states--fullwidth vertical">
    <scale-divider vertical data-sketch-key="divider-vertical"></scale-divider>
  </div>
</div>

<script src="/symbol_names.js"></script>
<script>
  Array.from(document.querySelectorAll('scale-divider')).forEach((btn) => {
    btn.dataset.sketchSymbol = `Divider / ${order(
      2,
      btn.getAttribute('vertical') === null ? 'Standard' : 'Vertical'
    )}`;
  });
</script>
