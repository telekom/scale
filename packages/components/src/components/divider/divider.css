/**
 * @license
 * Scale https://github.com/telekom/scale
 *
 * Copyright (c) 2021 Egor Kirpichev and contributors, Deutsche Telekom AG
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

:host {
  --width: 100%;
  --height: 100%;
  --spacing: var(--telekom-spacing-unit-x-3);
  /* FIXME:TOKEN this used to be scl-color-grey-20 */
  --color: var(--telekom-color-ui-subtle);
  --border-width: var(--telekom-spacing-unit-x-025);
  --min-height-vertical: var(--telekom-spacing-unit-x-6);

  width: var(--width);
  height: var(--height);
}

.divider {
  padding: var(--spacing);
}

.divider--vertical {
  display: inline-flex;
  height: inherit;
}

.divider__horizontal {
  margin: 0;
  border: 0;
  border-top: var(--border-width) solid var(--color);
}

.divider__vertical {
  display: inline-flex;
  height: inherit;
  min-height: var(--min-height-vertical);
  border-left: var(--border-width) solid var(--color);
}
