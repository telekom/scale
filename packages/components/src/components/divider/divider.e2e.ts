import { newE2EPage } from '@stencil/core/testing';

describe('t-divider', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-divider/>');
    const element = await page.find('t-divider');
    expect(element).toHaveClass('hydrated');
  });
});
