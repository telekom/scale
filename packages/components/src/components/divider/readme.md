# t-divider



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                 | Type      | Default |
| ------------- | -------------- | --------------------------- | --------- | ------- |
| `customClass` | `custom-class` | (optional) Divider class    | `string`  | `''`    |
| `size`        | `size`         | (optional) Divider size     | `string`  | `''`    |
| `theme`       | `theme`        | (optional) Divider theme    | `string`  | `''`    |
| `vertical`    | `vertical`     | (optional) Divider vertical | `boolean` | `false` |


## CSS Custom Properties

| Name                            | Description                         |
| ------------------------------- | ----------------------------------- |
| `--divider-border-top`          | Border top of the divider           |
| `--divider-color`               | Color of the divider                |
| `--divider-vertical-border`     | Border left of the vertical divider |
| `--divider-vertical-min-height` | Min height of the vertical divider  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
