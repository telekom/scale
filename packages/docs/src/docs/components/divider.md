---
title: Divider
description: Divider element
url: /docs/divider
contributors:
  - SylviaGonschior
---

# Divider

Divider component

selector: `t-divider`

## Default

```javascript
<t-divider></t-divider>
```

<div class="demo-container">
  <t-divider></t-divider>
</div>

## Vertical

```javascript
<t-divider horizontal="false"></t-divider>
```

<div class="demo-container">
  <t-divider horizontal="false"></t-divider>
</div>
