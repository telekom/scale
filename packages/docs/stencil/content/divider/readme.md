# scale-divider



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Divider class       | `string`                                 | `''`        |
| `size`        | `size`         | (optional) Divider size        | `string`                                 | `''`        |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |
| `vertical`    | `vertical`     | (optional) Divider vertical    | `boolean`                                | `false`     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
