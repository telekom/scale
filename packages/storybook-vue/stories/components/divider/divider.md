<div style="display: inline-flex; align-items: center; justify-content: space-between; width: 100%;">
    <h1>Divider</h1>
    <img src="assets/aa.png" alt="Accessible AA" />
</div>

A divider is a thin line that helps divide elements into groups. This makes it easy for users to identify content that belongs together.

## General

Dividers have a subtle visual effect â€“ visible but never dominant.

### When to use

Only use a divider:

- when using whitespace is not sufficient to create a content group or
- when using whitespace is not possible, e.g., when there's not enough space.

### When not to use

Only use a divider when really necessary. To avoid clutter and fragmentation, use as few dividers as possible.

## Variants

![Image Name](assets/3_components/divider/divider.png)
