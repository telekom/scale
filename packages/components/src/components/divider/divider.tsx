import { Component, Prop, h, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './divider.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

@Component({
  tag: 'scale-divider',
  shadow: true,
})
export class Divider implements Base {
  /** (optional) Divider class */
  @Prop() customClass?: string = '';
  /** (optional) Divider size */
  @Prop() size?: string = '';
  /** (optional) Divider vertical */
  @Prop() vertical?: boolean = false;

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Divider', styles) stylesheet: StyleSheet;

  componentWillUpdate() {}
  componentDidUnload() {}

  render() {
    const { classes } = this.stylesheet;
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <div class={this.getCssClassMap()}>
          {!this.vertical ? <hr /> : <span class={classes.divider__vertical} />}
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.divider,
      this.customClass && this.customClass,
      this.size && classes[`divider--size-${this.size}`],
      this.vertical && classes[`divider--vertical`]
    );
  }
}
