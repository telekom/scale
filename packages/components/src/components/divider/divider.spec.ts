import { newSpecPage } from '@stencil/core/testing';
import { Divider } from './divider';

describe('Divider', () => {
  let element;
  beforeEach(async () => {
    element = new Divider();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<t-divider></t-divider>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot when horizontal false', async () => {
    const page = await newSpecPage({
      components: [Divider],
      html: `<t-divider vertical=true></t-divider>`,
    });
    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should handle custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should handle vertical css class', () => {
    element.vertical = true;
    expect(element.getCssClassMap()).toContain('divider--vertical');
  });

  it('should handle default css class', () => {
    element.vertical = false;
    expect(element.getCssClassMap()).toContain('divider');
  });

  it('should handle size css class', () => {
    element.size = 'small';
    expect(element.getCssClassMap()).toContain('divider--size-small');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('divider--theme-default');
  });
});
