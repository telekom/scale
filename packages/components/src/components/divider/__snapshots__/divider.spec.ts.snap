// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should match snapshot 1`] = `
<scale-divider>
  <mock:shadow-root>
    <style>
      .\\:host-0-0-5 {
  width: 100%;
  height: 100%;
}
.divider-0-0-6 {}
.divider-0-0-6 hr {
  border: 0;
  border-top: 1px solid #dfdfdf;
}
.divider__vertical-0-0-7 {
  min-height: 1rem;
  border-left: 1px solid #dfdfdf;
}
.divider--vertical-0-0-8 {}
.divider--vertical-0-0-8, .divider--vertical-0-0-8 divider__vertical {
  height: inherit;
  display: inline-flex;
}
    </style>
    <div class="divider-0-0-6">
      <hr>
    </div>
  </mock:shadow-root>
</scale-divider>
`;

exports[`Divider should match snapshot when horizontal false 1`] = `
<scale-divider vertical="true">
  <mock:shadow-root>
    <style>
      .\\:host-0-0-13 {
  width: 100%;
  height: 100%;
}
.divider-0-0-14 {}
.divider-0-0-14 hr {
  border: 0;
  border-top: 1px solid #dfdfdf;
}
.divider__vertical-0-0-15 {
  min-height: 1rem;
  border-left: 1px solid #dfdfdf;
}
.divider--vertical-0-0-16 {}
.divider--vertical-0-0-16, .divider--vertical-0-0-16 divider__vertical {
  height: inherit;
  display: inline-flex;
}
    </style>
    <div class="divider--vertical-0-0-16 divider-0-0-14">
      <span class="divider__vertical-0-0-15"></span>
    </div>
  </mock:shadow-root>
</scale-divider>
`;
